cmd/convertor/builder/overlaybd_builder.go

```diff
@@ -135,6 +131,58 @@ func (e *overlaybdBuilderEngine) Cleanup() {
 	os.RemoveAll(e.workDir)
 }
 
+func (e *overlaybdBuilderEngine) uploadBaseLayer(ctx context.Context) (specs.Descriptor, error) {
+	// add baselayer with tar header
+	tarFile := path.Join(e.workDir, "ext4_64.tar")
+	fdes, err := os.Create(tarFile)
+	if err != nil {
+		return specs.Descriptor{}, errors.Wrapf(err, "failed to create file %s", tarFile)
+	}
+	digester := digest.Canonical.Digester()
+	countWriter := &writeCountWrapper{w: io.MultiWriter(fdes, digester.Hash())}
+	tarWriter := tar.NewWriter(countWriter)
```

Comment:
I recommand using the two functions `buildArchiveFromFiles()` and `getFileDesc()` in the `builder_utils.go` file to add tar header for baselayer, which makes the code simpler.

Reply:
UPDATE: We found `buildArchiveFromFiles()` may affect reproduciable for convertion, so we decided not to make this change for now. We will deal with this in a later commit.